codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -155,13 +154,18 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         }
     }
 
-    private void consumeBytes(int streamId, int bytes) {
+    private void consumeBytes(int streamId, int bytes, ChannelPromise promise) {
         try {
             Http2Stream stream = http2Handler.connection().stream(streamId);
             http2Handler.connection().local().flowController()
                         .consumeBytes(stream, bytes);
         } catch (Throwable t) {
+            promise.setFailure(t);
             exceptionCaught(ctx, t);
```

Comment:
@buchgr should we remove the `exceptionCaught(...)` here ? I think if we propagate to the promise we should not also call `exceptionCaught(...)`. Also @nmittler @Scottmitch WDYT ?